### PR TITLE
Fix save for CNN and BiLSTM in train

### LINKED ISSUE #32: Save throws error for CNN and BiLSTM
When trying to pickle a pipeline taht contains CNNClassifier or BiLSTMClassifier I get the following error
```
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/nicksorros/code/grants_tagger/grants_tagger/train.py", line 348, in <module>
    train_and_evaluate(
  File "/Users/nicksorros/code/grants_tagger/grants_tagger/train.py", line 275, in train_and_evaluate
    pickle.dump(model, f)
TypeError: cannot pickle '_thread.RLock' object
```

### LINKED ISSUE #33: Config 2020.04.2 and 2020.04.3 fail in save stage
This is related to #32 since these configs use the CNN and BiLSTM architectures

### LINKED ISSUE #34: y_batch_size fails when not provided
Currently we convert `y_batch_size` to into regardless whether it exists or is None. If None it currently fails.

### PR DESCRIPTION
**Pull request by nsorros**
Fixes #32. This the problem of not being able to pickle CNNClassifier and BiLSTMClassifier. 
Fixes #34. The above problem was affecting some prior configs in the last stage of saving.
Fixes #33 . A recently introduced feature `y_batch_size` has a bug when not provided

### CODE REVIEW

File: grants_tagger/train.py [hunk 1]
```diff
@@ -275,10 +275,22 @@ def train_and_evaluate(
...
+            if approach in ["cnn", "bilstm"]:
+                # cnn and bilstm are pipelines so do not have a save
+                # but CNNClassifier and BiLSTMClassifier do have one
+                vectorizer = model.steps[0][1]
+                classifier = model.steps[1][1]
+
+                vectorizer_path = os.path.join(model_path, "vectorizer.pkl")
+                with open(vectorizer_path, "wb") as f:
+                    f.write(pickle.dumps(vectorizer))
+                classifier.save(model_path)
```
Comment (lizgzil) [inline grants_tagger/train.py:290]: is the classifier save method ok to have a '.pkl' at the end? Usually it's input is just a directory right?

**Comment by lizgzil** [comment]
is there something weird about this branch? I've git fetched but cant switch to this branch?

**Review by lizgzil — COMMENTED** [review]
I think you'll see but my `make test` failed on
```
 classifier.save(model_path)
AttributeError: 'CNNClassifier' object has no attribute 'save'
```

**Comment by nsorros** [comment]
> I think you'll see but my `make test` failed on
> 
> ```
>  classifier.save(model_path)
> AttributeError: 'CNNClassifier' object has no attribute 'save'
> ```

Fixed, can you have another look?

**Review by lizgzil — APPROVED** [review]
Looks good. `make test` passes now

**Pull request merged**